c2rust-analyze/src/c_void_casts.rs

```diff
@@ -320,41 +405,44 @@ impl<'tcx> CVoidCasts<'tcx> {
             };
             let func_ty = func.ty(&body.local_decls, tcx);
 
-            // For [`CVoidCastDirection::From`], we only count
-            // a cast from `*c_void` to an arbitrary type in the subsequent block,
-            // searching forward.
-            let find_first_cast_succ_block = |get_cast| {
-                let successor_block_id = target.unwrap();
-                body.basic_blocks()[successor_block_id]
-                    .statements
-                    .iter()
-                    .find_map(get_cast)
-            };
-
-            // For [`CVoidCastDirection::From`], we only count
-            // a cast to `*c_void` from an arbitrary type in the same block,
-            // searching backwards.
-            let find_last_cast_curr_block =
-                |get_cast| bb_data.statements.iter().rev().find_map(get_cast);
-
             for direction in CVoidCastDirection::from_callee(ty_callee(tcx, func_ty))
                 .iter()
                 .copied()
             {
                 use CVoidCastDirection::*;
-                let c_void_ptr = match direction {
+                let c_void_ptr_place = match direction {
                     From => destination,
                     To => args[0].place().unwrap(),
                 };
-                let c_void_ptr = CVoidPtr::checked(c_void_ptr, &body.local_decls, tcx);
+                let c_void_ptr = CVoidPtr::checked(c_void_ptr_place, &body.local_decls, tcx);
```

Comment:
This was supposed to shadow the first `c_void_ptr`, now `c_void_ptr_place`.  Why the change?

Reply:
f6741d57